From a2b9afab8ed37b10d786ea7945df7194cd03eb06 Mon Sep 17 00:00:00 2001
From: raphielshira <raphielshira@neverland.localdomain>
Date: Mon, 22 Feb 2021 12:41:04 +0700
Subject: [PATCH] Log.cpp: Add missing mode
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This fixes the following build error:

In file included from /usr/include/fcntl.h:329,
                 from ../src/oomd/Log.cpp:20:
In function ‘int open(const char*, int, ...)’,
    inlined from ‘static bool Oomd::Log::init(const string&)’ at ../src/oomd/Log.cpp:76:23:
/usr/include/bits/fcntl2.h:50:24: error: call to ‘__open_missing_mode’ declared with attribute error: open with O_CREAT or O_TMPFILE in second argument needs 3 arguments
   50 |    __open_missing_mode ();
      |    ~~~~~~~~~~~~~~~~~~~~^~

Signed-off-by: raphielshira <raphielshira@neverland.localdomain>
---
 src/oomd/Log.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/oomd/Log.cpp b/src/oomd/Log.cpp
index b64f78f..b376dd1 100644
--- a/src/oomd/Log.cpp
+++ b/src/oomd/Log.cpp
@@ -73,7 +73,7 @@ Log::~Log() {
 }
 
 bool Log::init(const std::string& kmsg_path) {
-  int kmsg_fd = ::open(kmsg_path.c_str(), O_WRONLY | O_CREAT | O_APPEND);
+  int kmsg_fd = ::open(kmsg_path.c_str(), O_WRONLY | O_CREAT | O_APPEND, 0600);
   if (kmsg_fd < 0) {
     perror("open");
     std::cerr << "Unable to open outfile " << kmsg_path << ", not logging\n";
-- 
2.30.1

